### ISSUE TITLE
Some more `qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.` errors

### ISSUE THREAD

**Issue by piegamesde** [NixOS/nixpkgs#100427]
**Describe the bug**

Some QT applications (I only tested Kile, but it's probably some more) are broken on latest master.

**To Reproduce**

```sh
NIX_PATH=nixpkgs=https://github.com/nixos/nixpkgs/archive/90321a7bb31f129ce6b0594fa5614c7a83c72528.tar.gz nixGLIntel nix-shell -p 'kile' --pure --run kile
```
Error message:

```
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.
```

**Additional context**

This is similar to #99937. Also some of the discussion in [#99956](https://github.com/NixOS/nixpkgs/pull/99956#issuecomment-707815553)

**Notify maintainers**

@FRidh 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.8.10-arch1-1, Arch Linux, noversion`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(user): `"home-manager, nixpkgs-21.03pre244772.502845c3e31"`
 - nixpkgs: `~/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by FRidh** [comment]
fixed on master with c5a41da563485db7e34475321e22d45941de23d0 and on release-20.09 with 5f9c8a1976b63fb31d280f9d9fef3c0e678d0bec.

I did notice the theme was not OK for me on master, so we may need to downgrade to Qt 5.12 there as well. Let me know if there are any more issues!

**Comment by piegamesde** [comment]
Thanks for the quick fix.
